### PR TITLE
update the translation of `Object.hasOwnProperty`

### PR DESCRIPTION
**Pull request by robert-jx**
### Description

update the translation of `Object.hasOwnProperty`
### Motivation

update the translation of `Object.hasOwnProperty`
### Additional details

none

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md [hunk 1]
```diff
@@ -5,111 +5,143 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
...
-o.propOne = null;
-o.hasOwnProperty('propOne'); // 返回 true
-o.propTwo = undefined;
-o.hasOwnProperty('propTwo'); // 返回 true
+const fruits = ["Apple", "Banana", "Watermelon", "Orange"];
+fruits.hasOwnProperty(3); // 返回 true ('Orange')
+fruits.hasOwnProperty(4); // 返回 false——未定义
 ```
 
+该方法在重新实现的对象或者使用 `Object.create(null)` 创建的对象（因为这些对象不是从 `Object.prototype` 继承的）中不可使用。下面给出了这些情况的示例。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md:41]:
```suggestion
如果在对象中重新实现了该方法，或者使用 `Object.create(null)` 创建了对象（这些对象不继承自 `Object.prototype`），则该方法将不可用。下面给出了这些情况的示例。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md [hunk 2]
```diff
@@ -5,111 +5,143 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
...
+"prop" in example; // 返回 true
+"toString" in example; // 返回 true
+"hasOwnProperty" in example; // 返回 true
 ```
 
-### 遍历一个对象的所有自身属性
+### 遍历对象的属性
 
-下面的例子演示了如何在遍历一个对象的所有属性时忽略掉继承属性，注意这里 {{jsxref("Statements/for...in", "for...in")}} 循环只会遍历可枚举属性，所以不应该基于这个循环中没有不可枚举的属性而得出 `hasOwnProperty` 是严格限制于可枚举项目的（如同 {{jsxref("Object.getOwnPropertyNames()")}}）。
+以下示例展示了如何在不执行继承属性的情况下迭代对象的可枚举属性。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md:84]:
```suggestion
以下示例展示了如何在忽略继承属性的情况下迭代一个对象的可枚举属性。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md [hunk 3]
```diff
@@ -5,111 +5,143 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
...
-    console.log(name); // toString or something else
+    console.log(`this is fog (${name}) for sure. Value: ${buz[name]}`);
+  } else {
+    console.log(name); // toString 或其他的方法等
   }
 }
 ```
 
-### 使用 `hasOwnProperty` 作为属性名
+请注意，{{jsxref("Statements/for...in", "for...in")}} 循环只迭代可枚举项：不存在从循环发出的不可枚举属性并不意味着 `hasOwnProperty` 本身被严格限制在可枚举项中（如 {{jsxref("Object.getOwnPropertyNames()")}}）。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md:100]:
```suggestion
请注意，{{jsxref("Statements/for...in", "for...in")}} 循环只迭代可枚举项：循环中没有发出非可枚举属性并不意味着 `hasOwnProperty` 本身被严格限制在可枚举项中（和 {{jsxref("Object.getOwnPropertyNames()")}} 一样）。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md [hunk 4]
```diff
@@ -5,111 +5,143 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
...
+const foo = {
+  hasOwnProperty() {
     return false;
   },
-  bar: 'Here be dragons'
+  bar: "Here be dragons",
 };
 
-foo.hasOwnProperty('bar'); // 始终返回 false
+foo.hasOwnProperty("bar"); // 重复实现经常返回 false
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md:114]:
```suggestion
foo.hasOwnProperty("bar"); // 该重新实现始终返回 false
```

File: files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md [hunk 5]
```diff
@@ -5,111 +5,143 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
...
+```
 
-// 如果担心这种情况，
-// 可以直接使用原型链上真正的 hasOwnProperty 方法
-({}).hasOwnProperty.call(foo, 'bar'); // true
+克服这个问题的推荐方法是使用 {{jsxref("Object.hasOwn()")}}（在支持它的浏览器中）。其他选项包括使用*外部的* `hasOwnProperty`：
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md:117]:
```suggestion
克服这个问题的推荐方法是，在支持的浏览器中使用 {{jsxref("Object.hasOwn()")}}。其他替代方案包括使用*外部的* `hasOwnProperty`：
```

**Review by JasonLamv-t — COMMENTED** [review]
Some suggestions.

File: files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md [hunk 6]
```diff
@@ -5,111 +5,143 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
...
+```js-nolint
+hasOwnProperty(prop)
 ```
 
 ### 参数
 
-- _prop_
-  - : 要检测的属性的 {{jsxref("String")}} 字符串形式表示的名称，或者 {{jsxref("Symbol")}}。
+- `prop`
+  - : 要测试的属性的{{jsxref("String","字符串")}}名称或者 [Symbol](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Symbol)。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md:23]:
```suggestion
  - : 要测试的属性的{{jsxref("String","字符串", "", 1)}}名称或者 [Symbol](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Symbol)。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md [hunk 7]
```diff
@@ -5,111 +5,143 @@ slug: Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
...
-    console.log(name); // toString or something else
+    console.log(`this is fog (${name}) for sure. Value: ${buz[name]}`);
+  } else {
+    console.log(name); // toString 或其他的方法等
   }
 }
 ```
 
-### 使用 `hasOwnProperty` 作为属性名
+请注意，{{jsxref("Statements/for...in", "for...in")}} 循环只迭代可枚举项：循环中没有发出非可枚举属性并不意味着 `hasOwnProperty` 本身被严格限制在可枚举项中（和 {{jsxref("Object.getOwnPropertyNames()")}} 一样）。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/hasownproperty/index.md:100]:
```suggestion
请注意，{{jsxref("Statements/for...in", "for...in")}} 循环只迭代可枚举属性：循环中没有发出非可枚举属性并不意味着 `hasOwnProperty` 本身被严格限制在可枚举属性中（和 {{jsxref("Object.getOwnPropertyNames()")}} 一样）。
```

**Review by yin1999 — APPROVED** [review]

**Pull request merged**